block-producer/src/txqueue/mod.rs

```diff
@@ -158,10 +161,13 @@ where
     TV: TransactionVerifier,
     BB: BatchBuilder,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip_all, err(Debug), fields(COMPONENT = crate::COMPONENT))]
     async fn add_transaction(
```

Comment:
At the end of this function, after we add the transaction to the queue, would be great to print out total number of transaction in the queue.